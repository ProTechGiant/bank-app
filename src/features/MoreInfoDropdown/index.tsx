/* eslint-disable react-native/sort-styles */
import { useState } from "react";
import { Pressable, StyleSheet, View } from "react-native";

import { Icons } from "@/assets/icons";
import { iconDimensions, palette, radii, spacing } from "@/theme/values";

import Typography from "../../components/Typography";
import { GreyGradient } from "../../components/LinearGradients/GradientButtons";
import Dropdown from "../../components/Dropdown";

interface MoreInfoDropdownProps {
  title: string;
  children: JSX.Element | JSX.Element[];
}

const MoreInfoDropdown = ({ title, children }: MoreInfoDropdownProps) => {
  // const ChevronUp = Icons["ChevronUp"];
  const DownArrow = Icons.DownArrow;
  const UpArrow = Icons.UpArrow;
  const InfoCircle = Icons["InfoCircle"];

  const [openContent, setOpenContent] = useState<boolean>(false);
  const toggleContent = () => {
    setOpenContent(!openContent);
  };

  return (
    <View style={styles.container}>
      <Dropdown
        title={
          <View style={styles.toggleContainer}>
            <GreyGradient>
              <Pressable onPress={toggleContent}>
                <View style={styles.pressableContainer}>
                  <View style={styles.infoIcon}>
                    <InfoCircle />
                  </View>
                  <Typography.Text weight="semiBold" size="callout" style={styles.title}>
                    {title}
                  </Typography.Text>
                  <View style={styles.toggleIcon}>
                    {openContent ? (
                      <UpArrow color={palette.primaryBase} width={iconDimensions.dropdown} />
                    ) : (
                      <DownArrow color={palette.primaryBase} width={iconDimensions.dropdown} />
                    )}
                  </View>
                </View>
              </Pressable>
            </GreyGradient>
          </View>
        }
        dropdownVisible={openContent}
        content={<View style={styles.contentContainer}>{children}</View>}
      />
    </View>
  );
};

export default MoreInfoDropdown;

const styles = StyleSheet.create({
  container: {
    width: "100%",
    backgroundColor: palette["neutralBase-50"],
    borderRadius: radii.extraSmall,
  },
  contentContainer: {
    paddingHorizontal: 40,
    paddingVertical: spacing.medium,
  },
  pressableContainer: {
    padding: spacing.medium,
    flexDirection: "row",
    alignItems: "center",
    width: "100%",
  },
  toggleContainer: {
    borderRadius: radii.extraSmall,
    overflow: "hidden",
  },
  toggleIcon: {
    height: iconDimensions.accordian,
    width: iconDimensions.accordian,
    justifyContent: "center",
    alignItems: "flex-end",
  },
  infoIcon: {
    height: iconDimensions.accordian,
    width: iconDimensions.accordian,
    justifyContent: "center",
    alignItems: "flex-start",
    marginRight: spacing.small,
  },
  title: {
    flex: 1,
  },
});
