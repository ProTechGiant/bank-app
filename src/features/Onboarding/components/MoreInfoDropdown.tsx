import { useState } from "react";
import { Pressable, View, ViewStyle } from "react-native";

import { AngleDownIcon, AngleUpIcon, InfoCircleIcon } from "@/assets/icons";
import { GreyGradient } from "@/components/LinearGradients/GradientButtons";
import Typography from "@/components/Typography";
import { useThemeStyles } from "@/theme";

interface MoreInfoDropdownProps {
  children: JSX.Element | JSX.Element[];
  title: string;
}

export default function MoreInfoDropdown({ children, title }: MoreInfoDropdownProps) {
  const [isExpanded, setIsExpanded] = useState(false);

  const containerStyle = useThemeStyles<ViewStyle>(theme => ({
    backgroundColor: theme.palette["neutralBase-50"],
    borderRadius: theme.radii.extraSmall,
    overflow: "hidden",
    width: "100%",
  }));

  const contentStyle = useThemeStyles<ViewStyle>(theme => ({
    padding: theme.spacing["16p"],
  }));

  const pressableContainerStyle = useThemeStyles<ViewStyle>(theme => ({
    padding: theme.spacing["16p"],
    flexDirection: "row",
    alignItems: "center",
    width: "100%",
    backgroundColor: theme.palette["neutralBase-60"],
  }));

  const infoIconColor = useThemeStyles(theme => theme.palette["primaryBase-40"]);

  const toggleIconStyle = useThemeStyles<ViewStyle>(theme => ({
    height: theme.iconDimensions.accordian,
    width: theme.iconDimensions.accordian,
    justifyContent: "center",
    alignItems: "flex-end",
  }));

  const titleStyles = useThemeStyles<ViewStyle>(theme => ({
    flexGrow: 1,
    marginHorizontal: theme.spacing["8p"],
  }));

  return (
    <Pressable onPress={() => setIsExpanded(c => !c)} style={containerStyle}>
      <GreyGradient>
        <View style={pressableContainerStyle}>
          <InfoCircleIcon color={infoIconColor} />
          <Typography.Text color="primaryBase" weight="semiBold" size="footnote" style={titleStyles}>
            {title}
          </Typography.Text>
          <View style={toggleIconStyle}>{isExpanded ? <AngleUpIcon /> : <AngleDownIcon />}</View>
        </View>
      </GreyGradient>
      {isExpanded && <View style={contentStyle}>{children}</View>}
    </Pressable>
  );
}
