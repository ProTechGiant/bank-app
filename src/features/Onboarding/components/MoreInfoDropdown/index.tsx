import { useState } from "react";
import { Pressable, View, ViewStyle } from "react-native";

import { AngleDownIcon, AngleUpIcon, InfoCircleIcon } from "@/assets/icons";
import { GreyGradient } from "@/components/LinearGradients/GradientButtons";
import Typography from "@/components/Typography";
import { useThemeStyles } from "@/theme";

interface MoreInfoDropdownProps {
  children: JSX.Element | JSX.Element[];
  title: string;
}

export default function MoreInfoDropdown({ children, title }: MoreInfoDropdownProps) {
  const [isExpanded, setIsExpanded] = useState(false);

  const containerStyle = useThemeStyles<ViewStyle>(theme => ({
    backgroundColor: theme.palette["neutralBase-50"],
    borderRadius: theme.radii.extraSmall,
    overflow: "hidden",
    width: "100%",
  }));

  const contentStyle = useThemeStyles<ViewStyle>(theme => ({
    paddingHorizontal: theme.spacing["32p"],
    paddingVertical: theme.spacing["16p"],
  }));

  const pressableContainerStyle = useThemeStyles<ViewStyle>(theme => ({
    padding: theme.spacing["16p"],
    flexDirection: "row",
    alignItems: "center",
    width: "100%",
  }));

  const toggleIconStyle = useThemeStyles<ViewStyle>(theme => ({
    height: theme.iconDimensions.accordian,
    width: theme.iconDimensions.accordian,
    justifyContent: "center",
    alignItems: "flex-end",
  }));

  const titleStyles = useThemeStyles<ViewStyle>(theme => ({
    flexGrow: 1,
    marginHorizontal: theme.spacing["8p"],
  }));

  return (
    <Pressable onPress={() => setIsExpanded(c => !c)} style={containerStyle}>
      <GreyGradient>
        <View style={pressableContainerStyle}>
          <InfoCircleIcon />
          <Typography.Text color="primaryBase" weight="semiBold" size="callout" style={titleStyles}>
            {title}
          </Typography.Text>
          <View style={toggleIconStyle}>{isExpanded ? <AngleUpIcon /> : <AngleDownIcon />}</View>
        </View>
      </GreyGradient>
      {isExpanded && <View style={contentStyle}>{children}</View>}
    </Pressable>
  );
}
