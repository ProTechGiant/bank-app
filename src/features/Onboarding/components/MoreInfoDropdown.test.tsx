import { fireEvent, render, screen } from "@testing-library/react-native";
import { Text } from "react-native";
import { create } from "react-test-renderer";

import MoreInfoDropdown from "./MoreInfoDropdown";

describe("components/MoreInfoDropdown", () => {
  it("renders a MoreInfoDropdown", () => {
    const result = create(
      <MoreInfoDropdown title="TitleTest">
        <Text>BodyTest</Text>
      </MoreInfoDropdown>
    );

    expect(result).toMatchSnapshot();
  });
  it("opens dropdown on pressable", () => {
    const { getByText, queryByText } = render(
      <MoreInfoDropdown title="TestTitle">
        <Text>Body Content Test</Text>
      </MoreInfoDropdown>
    );

    const element = getByText("TestTitle");
    const initialContents = queryByText("Body Content Test");

    expect(initialContents).toBeNull();

    fireEvent.press(element);

    const openContents = queryByText("Body Content Test");
    expect(openContents).not.toBeNull();

    fireEvent.press(element);

    expect(initialContents).toBeNull();
  });
});
