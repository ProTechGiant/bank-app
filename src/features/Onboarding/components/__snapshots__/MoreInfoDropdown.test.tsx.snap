// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MoreInfoDropdown renders a MoreInfoDropdown 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 4,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 4,
          "overflow": "hidden",
        }
      }
    >
      <BVLinearGradient
        colors={
          Array [
            4292403950,
            4292927977,
          ]
        }
        endPoint={
          Object {
            "x": 1,
            "y": 1,
          }
        }
        locations={null}
        startPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 16,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "height": 24,
                  "justifyContent": "center",
                  "marginRight": 8,
                  "width": 24,
                }
              }
            >
              <SvgMock />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Object {
                    "color": "#080E53",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "lineHeight": 21,
                  },
                ]
              }
            >
              TitleTest
            </Text>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "height": 24,
                  "justifyContent": "center",
                  "width": 24,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="10"
                bbWidth="15"
                fill="none"
                focusable={false}
                height={10}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 10,
                      "width": 15,
                    },
                  ]
                }
                vbHeight={8}
                vbWidth={14}
                width={15}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M7.00367 7.58231C7.28422 7.58231 7.51687 7.47967 7.73584 7.2607L12.8884 1.9918C13.0663 1.81389 13.1484 1.60176 13.1484 1.34858C13.1484 0.835375 12.7447 0.417969 12.2315 0.417969C11.9783 0.417969 11.7457 0.527452 11.5609 0.712206L7.01051 5.39263L2.44641 0.712206C2.26165 0.527452 2.03584 0.417968 1.77582 0.417968C1.26262 0.417968 0.852051 0.835375 0.852051 1.34858C0.852051 1.60176 0.941007 1.81389 1.11892 1.9918L6.27149 7.2607C6.49046 7.48651 6.72311 7.58231 7.00367 7.58231Z"
                    fill={
                      Object {
                        "payload": 4278718035,
                        "type": 0,
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </BVLinearGradient>
    </View>
  </View>
</View>
`;
